Bump nan to 1.6 and fix compilation issue

Currenlty project relies on nan at v1.5. It would be good to upgrade it so v1.6 is used.

At current point just bumping dependency is not enough, as there's below compilation error:

```
> libxmljs@0.13.0 install /Users/medikoo/Projects/eregistrations/lomas/node_modules/invalidatejs/node_modules/libxml-to-js/node_modules/libxmljs
> node-gyp rebuild

child_process: customFds option is deprecated, use stdio instead.
  CC(target) Release/obj.target/libxml/vendor/libxml/buf.o
  CC(target) Release/obj.target/libxml/vendor/libxml/catalog.o
  CC(target) Release/obj.target/libxml/vendor/libxml/chvalid.o
  CC(target) Release/obj.target/libxml/vendor/libxml/dict.o
  CC(target) Release/obj.target/libxml/vendor/libxml/encoding.o
  CC(target) Release/obj.target/libxml/vendor/libxml/entities.o
  CC(target) Release/obj.target/libxml/vendor/libxml/error.o
  CC(target) Release/obj.target/libxml/vendor/libxml/globals.o
  CC(target) Release/obj.target/libxml/vendor/libxml/hash.o
  CC(target) Release/obj.target/libxml/vendor/libxml/HTMLparser.o
  CC(target) Release/obj.target/libxml/vendor/libxml/HTMLtree.o
  CC(target) Release/obj.target/libxml/vendor/libxml/legacy.o
  CC(target) Release/obj.target/libxml/vendor/libxml/list.o
  CC(target) Release/obj.target/libxml/vendor/libxml/parser.o
../vendor/libxml/parser.c:13492:43: warning: passing 'const xmlChar *' (aka 'const unsigned char *') to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        hdlr = xmlFindCharEncodingHandler(doc->encoding);
                                          ^~~~~~~~~~~~~
../vendor/libxml/include/libxml/encoding.h:175:42: note: passing argument to parameter 'name' here
        xmlFindCharEncodingHandler      (const char *name);
                                                     ^
1 warning generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/parserInternals.o
  CC(target) Release/obj.target/libxml/vendor/libxml/pattern.o
  CC(target) Release/obj.target/libxml/vendor/libxml/relaxng.o
  CC(target) Release/obj.target/libxml/vendor/libxml/SAX2.o
  CC(target) Release/obj.target/libxml/vendor/libxml/SAX.o
  CC(target) Release/obj.target/libxml/vendor/libxml/tree.o
  CC(target) Release/obj.target/libxml/vendor/libxml/threads.o
  CC(target) Release/obj.target/libxml/vendor/libxml/uri.o
  CC(target) Release/obj.target/libxml/vendor/libxml/valid.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xinclude.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xlink.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlIO.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlmemory.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlmodule.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlreader.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlregexp.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlsave.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlschemas.o
../vendor/libxml/xmlschemas.c:24203:10: warning: explicitly assigning a variable of type 'xmlSchemaValType' to itself [-Wself-assign]
        valType = valType;
        ~~~~~~~ ^ ~~~~~~~
1 warning generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlschemastypes.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlstring.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlunicode.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlwriter.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xpath.o
../vendor/libxml/xpath.c:12402:13: warning: comparison of constant 'XML_NAMESPACE_DECL' (18) with expression of type 'xmlXPathTypeVal' is always false [-Wtautological-constant-out-of-range-compare]
                        if (type == XML_NAMESPACE_DECL)
                            ~~~~ ^  ~~~~~~~~~~~~~~~~~~
In file included from ../vendor/libxml/xpath.c:185:
../vendor/libxml/trionan.c:194:1: warning: unused function 'trio_is_special_quantity' [-Wunused-function]
trio_is_special_quantity
^
../vendor/libxml/trionan.c:218:1: warning: unused function 'trio_is_negative' [-Wunused-function]
trio_is_negative
^
3 warnings generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
  CXX(target) Release/obj.target/xmljs/src/xml_attribute.o
In file included from ../src/xml_attribute.cc:2:
In file included from ../src/xml_attribute.h:5:
In file included from ../src/libxmljs.h:7:
In file included from ../../../../../../nan/nan.h:74:
../../../../../../nan/nan_new.h:249:55: error: no viable conversion from 'void (*)(const v8::FunctionCallbackInfo<v8::Value> &)' to 'v8::Handle<v8::Signature>'
    return NanIntern::Factory<T>::New(callback, data, a2);
                                                      ^~
../src/xml_attribute.cc:138:4: note: in instantiation of function template specialization 'NanNew<v8::FunctionTemplate, void (*)(const v8::FunctionCallbackInfo<v8::Value> &)>' requested here
   NanNew<v8::FunctionTemplate, NanFunctionCallback>(XmlAttribute::New);
   ^
/Users/medikoo/.node-gyp/0.12.0/deps/v8/include/v8.h:196:26: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'void (*)(const v8::FunctionCallbackInfo<v8::Value> &)' to 'const v8::Handle<v8::Signature> &' for 1st argument
template <class T> class Handle {
                         ^
/Users/medikoo/.node-gyp/0.12.0/deps/v8/include/v8.h:213:32: note: candidate template ignored: could not match 'Handle<type-parameter-0-0>' against 'void (*)(const v8::FunctionCallbackInfo<v8::Value> &)'
  template <class S> V8_INLINE Handle(Handle<S> that)
                               ^
../../../../../../nan/nan_implementation_12_inl.h:89:63: note: passing argument to parameter 'signature' here
                                  , v8::Handle<v8::Signature> signature) {
                                                              ^
1 error generated.
make: *** [Release/obj.target/xmljs/src/xml_attribute.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1067:12)
gyp ERR! System Darwin 14.1.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/medikoo/Projects/eregistrations/lomas/node_modules/invalidatejs/node_modules/libxml-to-js/node_modules/libxmljs
gyp ERR! node -v v0.12.0
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 

npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/usr/local/bin/npm" "rebuild"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE
npm ERR! libxmljs@0.13.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the libxmljs@0.13.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the libxmljs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls libxmljs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/medikoo/Projects/eregistrations/lomas/npm-debug.log
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,3 +3,7 @@ node_js:
     - "0.8"
     - "0.10"
     - "0.11"
+before_install:
+  - npm install npm
+  - mv node_modules npm
+  - npm/.bin/npm install

--- a/package.json
+++ b/package.json
@@ -21,8 +21,8 @@
     "node": ">=0.8.0"
   },
   "dependencies": {
-    "bindings": "1.1.1",
-    "nan": "1.5.1"
+    "bindings": "1.2.1",
+    "nan": "1.7.0"
   },
   "devDependencies": {
     "nodeunit": "0.9.0"

--- a/src/xml_attribute.cc
+++ b/src/xml_attribute.cc
@@ -151,7 +151,7 @@ void
 XmlAttribute::Initialize(v8::Handle<v8::Object> target) {
   NanScope();
   v8::Local<v8::FunctionTemplate> tmpl =
-   NanNew<v8::FunctionTemplate, NanFunctionCallback>(XmlAttribute::New);
+   NanNew<v8::FunctionTemplate>(XmlAttribute::New);
   NanAssignPersistent(constructor_template, tmpl);
   tmpl->Inherit(NanNew(XmlNode::constructor_template));
   tmpl->InstanceTemplate()->SetInternalFieldCount(1);

--- a/src/xml_document.cc
+++ b/src/xml_document.cc
@@ -450,7 +450,7 @@ XmlDocument::Initialize(v8::Handle<v8::Object> target)
     NanScope();
 
     v8::Local<v8::FunctionTemplate> tmpl =
-      NanNew<v8::FunctionTemplate, NanFunctionCallback>(New);
+      NanNew<v8::FunctionTemplate>(New);
     tmpl->SetClassName(NanNew<v8::String>("Document"));
 
     NanAssignPersistent(constructor_template, tmpl);

--- a/src/xml_element.cc
+++ b/src/xml_element.cc
@@ -492,7 +492,7 @@ XmlElement::Initialize(v8::Handle<v8::Object> target)
 {
     NanScope();
     v8::Local<v8::FunctionTemplate> tmpl =
-      NanNew<v8::FunctionTemplate, NanFunctionCallback>(New);
+      NanNew<v8::FunctionTemplate>(New);
     NanAssignPersistent(constructor_template, tmpl);
     tmpl->Inherit(NanNew(XmlNode::constructor_template));
     tmpl->InstanceTemplate()->SetInternalFieldCount(1);

--- a/src/xml_namespace.cc
+++ b/src/xml_namespace.cc
@@ -127,7 +127,7 @@ void
 XmlNamespace::Initialize(v8::Handle<v8::Object> target) {
   NanScope();
   v8::Local<v8::FunctionTemplate> tmpl =
-    NanNew<v8::FunctionTemplate, NanFunctionCallback>(New);
+    NanNew<v8::FunctionTemplate>(New);
   NanAssignPersistent(constructor_template, tmpl);
   tmpl->InstanceTemplate()->SetInternalFieldCount(1);
 

